lib/portlayer/storage/nfs/target.go

```diff
@@ -0,0 +1,50 @@
+// Copyright 2016-2017 VMware, Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package nfs
+
+import (
+	"io"
+	"net/url"
+
+	"os"
+)
+
+type MountHandler interface {
+	Mount(target *url.URL) (NFSTarget, error)
+	Unmount(target NFSTarget) error
+}
+
+type NFSTarget interface {
```

Comment:
This seems generic enough that it could just be `FSTarget`, or even just `FileSystem`. You could even argue some of the functionality is part of https://github.com/vmware/vic/blob/master/lib/system/syscall.go